src/lib.rs

```diff
@@ -1277,6 +1277,44 @@ impl Error {
         self.depth
     }
 
+    /// Inspect the underlying [`io::Error`] if there is one.
+    /// If a cycle was detected instead, [`None`] is returned.
```

Comment:
I would just say here that if the underlying error isn't an `io::Error` than the result is `None`. My other comment might make this obsolete though.